include/irr/asset/CTexturePacker.h

```diff
@@ -0,0 +1,107 @@
+#ifndef __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+#define __IRR_C_TEXTURE_PACKER_H_INCLUDED__
+
+#include "irr/asset/format/EFormat.h"
+#include "irr/asset/ICPUImage.h"
+#include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
+
+namespace irr {
+namespace asset
+{
+
+class CTexturePacker
+{
+    using addr_alctr_t = core::GeneralpurposeAddressAllocator<uint32_t>;
+
+public:
+    struct STexOffset
+    {
+        uint32_t x, y;
+    };
+
+    CTexturePacker(E_FORMAT _format, uint32_t _szxy = 1u<<14, uint32_t _pgSzxy = 256u) :
```

Comment:
no need for an `E_FORMAT` you'd just need an `E_FORMAT_CLASS` (one enum for all binary compatible formats)

You need to decide if `_szxy` means all the pixels used by the texture or just along one dimension, because the way you've passed it to `reserved_size` implies that its actually the whole texture.

I'd rather specify arguments the user can't "fuck up", so
`uint32_t pageCountBits=14u, uint32_t pageSizeBitsInTexelBlocks=8u`

then page size in texel blocks : `1u<<pageSizeBitsInTexelBlocks==256u`
then total bytes under managment: `asset::getTexetBlockSize(_format_class)<<(pageCountBits+pageSizeBitsInTexelBlocks)`